lib/types.d.ts

```diff
@@ -0,0 +1,52 @@
+declare module 'json5' {
+  type StringifyOptions = Partial<{
+    /**
+     * A function that alters the behavior of the stringification process, or an array of String and Number objects that serve as a whitelist for selecting/filtering the properties of the value object to be included in the JSON5 string. If this value is null or not provided, all properties of the object are included in the resulting JSON5 string.
+     */
+    replacer: (this: any, key: string, value: any) => any;
+
+    /**
+     * A String or Number object that's used to insert white space into the output JSON5 string for readability purposes. If this is a Number, it indicates the number of space characters to use as white space; this number is capped at 10 (if it is greater, the value is just 10). Values less than 1 indicate that no space should be used. If this is a String, the string (or the first 10 characters of the string, if it's longer than that) is used as white space. If this parameter is not provided (or is null), no white space is used. If white space is used, trailing commas will be used in objects and arrays.
+     */
+    space: string | number;
+
+    /**
+     * A String representing the quote character to use when serializing strings.
+     */
+    quote: string;
+  }>
+
+  /**
+   * Parses a JSON5 string, constructing the JavaScript value or object described
+   * by the string. An optional reviver function can be provided to perform a
+   * transformation on the resulting object before it is returned.
+   * @param text The string to parse as JSON5.
+   * @param reviver If a function, this prescribes how the value originally produced by parsing is transformed, before being returned.
+   */
+  export function parse(text: string, reviver?: (this: any, key: string, value: any) => any): any;
```

Comment:
Does it make sense to add a `parse<T>(...): T` overload if TypeScript users already know what they're expecting to get from `parse`? Or maybe it would just add confusion since users might expect it to convert the return value to `T`.

Reply:
+1 for `parse<T>(...): T`. Many people will know that `T` is only a type thing. The others will learn something :)

Reply:
Sorted. :+1: